ping.py

```diff
@@ -0,0 +1,26 @@
+import os
+import time
+
+from sqlalchemy import create_engine
+from sqlalchemy.engine.url import make_url
+
+from CTFd.config import Config
+
+url = make_url(Config.DATABASE_URL)
+if not url.drivername.startswith("mysql"):
+    exit(0)
+
+engine = create_engine(url)
+print(f"Waiting for {url.host}:{url.port or 3306} to be ready")
+
+while True:
+    try:
+        engine.raw_connection().ping()
+        print()
```

Comment:
I don't think this is necessary? Does it just mimic the original mysql client's behavior?

Reply:
hmmm... no it doesn't. I agree this is weird, logging in to the database *then* sending a ping packet. does removing the `.ping()` part sound reasonable? by doing so we can also remove the `driver.startswith(mysql)` check

Reply:
how about

```python
while true try:
    engine.connect()
except pymysql.err.Error:
    ...
```

Reply:
I just meant removing the empty `print` statement here actually but I will look into the `ping` method as well.

Reply:
Does this work though? I'm getting an error if MySQL isn't on. 

```
InternalError: (pymysql.err.InternalError) (1049, "Unknown database 'ctfd'")
(Background on this error at: http://sqlalche.me/e/2j85)
```

Reply:
We do have SQLAlchemy-Utils so maybe you can use https://sqlalchemy-utils.readthedocs.io/en/latest/database_helpers.html?highlight=database_exists#sqlalchemy_utils.functions.database_exists?

Reply:
It looks like you can do something like this: https://stackoverflow.com/a/17941078

Without a database then you can test for the raw connection.

Reply:
Looks like this strategy would also work for postgres. 

So you would just need to take the url and recreate it without the database.

Reply:
I wrote that on my phone, haven't tested, sorry😓
it came to my mind that mariadb image actually created the database for us
![image](https://user-images.githubusercontent.com/20221896/114655199-54aaee00-9d1e-11eb-9862-1dafb06f684e.png)
so.... `database_exists` is fine I think?

Reply:
You're right but `mysqladmin ping` only tests for connectivity to the database server. It doesn't check for existence of the database. It might be unnecessary but I would still rather let CTFd be responsible for creating the database if it doesn't exist. 

I think just creating a Python version of `mysqladmin ping` is the best way forward so we don't accidentally break something.

So just https://stackoverflow.com/a/17941078.

Reply:
I found that it's actually not that complicated

Reply:
the empty print statement is to print('ready') on a new line.

before:

```
Waiting for db: to be ready
..........db is ready
```

after:

```
Waiting for mysql+pymysql://ctfd:ctfd@db to be ready
.........
mysql+pymysql://ctfd:ctfd@db is ready
```

Reply:
I don't really think it's necessary to have. Can you remove it?